### PR TITLE
Ensure minimum mirror interval is reported on settings page

### LINKED ISSUE #3737: Error message should have given better error message on what is wrong at least if there is any minimum
- Gitea version (or commit ref): 321cc2a
- Git version: -
- Operating system: docker
- Database (use `[x]`):
  - [ x ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ x ] Not relevant
- Log gist:
https://gist.github.com/boskiv/cffeafcba8dc8b46fa7e095266f1ef25
## Description
Changing mirror interval to `5m` , raise non-informal message
Minimum available interval `10m`

## Screenshots

screen shot 2018-03-29 at 11 23 26

### PR DESCRIPTION
**Pull request by zeripath**
Expecting users to guess the minimum mirror interval appears a little unkind.
In this PR we simply change the locale string to include the minimum interval.
This will of course be affected by our current localization framework but...
we can fix that else where.

This PR also includes some fixes for error handling on the settings page as
previously the mirror block amongst others would simply disappear on error.

Fix #3737

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: routers/web/repo/setting.go [hunk 1]
```diff
@@ -102,6 +104,17 @@ func SettingsPost(ctx *context.Context) {
...
+	ctx.Data["MirrorsEnabled"] = setting.Mirror.Enabled
+	ctx.Data["DisableNewPushMirrors"] = setting.Mirror.DisableNewPush
+	ctx.Data["DefaultMirrorInterval"] = setting.Mirror.DefaultInterval
+	ctx.Data["MinimumMirrorInterval"] = setting.Mirror.MinInterval
+
+	signing, _ := asymkey_service.SigningKey(ctx, ctx.Repo.Repository.RepoPath())
+	ctx.Data["SigningKeyAvailable"] = len(signing) > 0
+	ctx.Data["SigningSettings"] = setting.Repository.Signing
+	ctx.Data["CodeIndexerEnabled"] = setting.Indexer.RepoIndexerEnabled
+
```
Comment (Gusted) [inline routers/web/repo/setting.go:117]: FYI this will conflict with #19828

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -860,7 +860,7 @@ default_branch = Default Branch
 default_branch_helper = The default branch is the base branch for pull requests and code commits.
 mirror_prune = Prune
 mirror_prune_desc = Remove obsolete remote-tracking references
-mirror_interval = Mirror Interval (valid time units are 'h', 'm', 's'). 0 to disable automatic sync.
+mirror_interval = Mirror Interval (valid time units are 'h', 'm', 's'). 0 to disable automatic sync. (Minimum interval: %v)
```
Comment (Gusted) [inline options/locale/locale_en-US.ini]:
```suggestion
mirror_interval = Mirror Interval (valid time units are 'h', 'm', 's'). 0 to disable automatic sync. (Minimum interval: %s)
```

time.Duration has a `String()` function, no need to use `%v`.

**Review by Gusted — APPROVED** [review]

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -1532,6 +1532,7 @@ pulls.data_broken = This pull request is broken due to missing fork information.
 pulls.files_conflicted = This pull request has changes conflicting with the target branch.
 pulls.is_checking = "Merge conflict checking is in progress. Try again in few moments."
 pulls.is_empty = "This branch is equal with the target branch."
+pulls.is_empty_differing_head_merge_base = "This branch ."
```
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: This line?

Comment (zeripath) [reply options/locale/locale_en-US.ini]:
```suggestion
```

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**